### ISSUE TITLE
error  Unsafe assignment of an `any` value

### ISSUE THREAD

**Issue by koji--** [i18next/i18next#2011]
## 💥 Regression Report

When update from v22.5.1 to v23.4.1, eslint gives title error.

When embedding a value in the second argument of the `t()` function, if there was a place where the `t()` function was used, it seemed to cause an error.

## Last working version

Worked up to version: 22.5.1

Stopped working in version: 23.4.1(Probably from 23.0.0)

## To Reproduce

I am trying in the following environment.

https://github.com/koji--/i18n-test

```
// is set to v23.4.1 and run eslint
$ yarn eslint
yarn run v1.22.19
$ eslint --ext .ts,tsx ./app

/path/to/i18n-test/app/app.tsx
  22:42  error  Unsafe assignment of an `any` value  @typescript-eslint/no-unsafe-assignment

✖ 1 problem (1 error, 0 warnings)

error Command failed with exit code 1.

// set to v22.5.1
$ yarn add i18next@22.5.1
...

// run eslint again
$ yarn eslint
yarn run v1.22.19
$ eslint --ext .ts,tsx ./app
✨  Done in 2.05s.
```

## Expected behavior

No errors in eslint like it did with v22.5.1.

## Your Environment

- *runtime version*: node v20.3.1
- *i18next version*: v23.4.1
- *os*: Mac

**Comment by adrai** [comment]
workaraound: 

```diff
- console.log("hoge: ", i18next.t("d", { val: i18next.t("a") }));
+ console.log("hoge: ", i18next.t("d", { val: i18next.t("a") as string }));
```

**Comment by koji--** [comment]
@adrai Thank you for reply. It's true that the errors can be resolved with that workaround, but what concerns me is whether the change from string type to any type aligns with the implementer's intention. If this is intentional, it's going to be tough, but I plan to make the necessary adjustments in the parts where i18next is being used. If it's not intentional and can be addressed on the i18next side, I intend to wait for that.

**Comment by adrai** [comment]
Need to wait for @pedrodurek feedback

**Comment by jwesselink** [comment]
What version of Typescript are you using? Using v5 fixed the error for me (see: https://github.com/i18next/i18next/issues/2008)

**Comment by koji--** [comment]
https://github.com/koji--/i18n-test/blob/main/package.json#L32

`typescript": "5.1.6`

**Comment by marcalexiei** [comment]
@koji--  This issue should be now fixed in [v23.5.0](https://github.com/i18next/i18next/releases/tag/v23.5.0).
I just finished trying it myself and everything works correctly.

**Comment by jbanulso** [comment]
I'm still experiencing the same ESLint error with v23.5.0. I realise this GitHub issue was originally about the second argument of the `t` function, and in my case is just about the `t` function itself, so please let me know if it'd be better to create a new GitHub issue.

I've tried to recreate a somewhat minimal repro here: https://codesandbox.io/s/pensive-feather-lwwpwz?file=/src/App.tsx:222-270. Unfortunately I haven't managed ESLint to bring the said error forward, but the question remains, why does the result from the `t` function have `any` type instead of `string`?

**Comment by koji--** [comment]
@marcalexiei Thank you. I also solved it with v23.5.0.

**Comment by koji--** [comment]
@jbanulso Actually, I'm having the same error with a different cause.

I'm going to see if I can create an issue for that, but it seems like it would be a good idea for you to create a separate issue as well.
